pkg/resource/deploy/plan.go

```diff
@@ -60,6 +64,36 @@ type PlanSummary interface {
 	Sames() map[resource.URN]bool
 }
 
+// InvalidResourceError is returned by deploy.NewPlan when the engine observes
+// that one or more resources are in an invalid state. The engine populates `InvalidResources`
+// with the URNs of every resource that was invalid.
+type InvalidResourceError struct {
+	InvalidResources []*resource.State
+}
+
+func (ire InvalidResourceError) Error() string {
+	var buf bytes.Buffer
+	writer := bufio.NewWriter(&buf)
+	fmt.Fprintf(writer,
+		"error: the current deployment has %d resource(s) whose statuses are unknown:\n", len(ire.InvalidResources))
+
+	for _, res := range ire.InvalidResources {
+		fmt.Fprintf(writer, "  * %s, interrupted while %s\n", res.URN, res.Status)
+	}
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "These resources have unknown statuses because the Pulumi CLI was interrupted before it\n")
+	fmt.Fprintf(writer, "had the opportunity to see if an operation that it initiated was successful. You should\n")
+	fmt.Fprintf(writer, "confirm whether or not the above operations completed successfully with your cloud provider.\n")
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "Once you have confirmed the status of the interrupted operations, you can repair your stack\n")
+	fmt.Fprintf(writer, "using `pulumi stack export` to export your stack to a file. For each operation that succeeded,\n")
+	fmt.Fprintf(writer, "remove the `status` field. Once this is complete, use `pulumi stack export` to import the\n")
```

Comment:
I think you mean import not export here?

Reply:
you're right, good catch!